File: android_fragment/commons-core/src/main/java/com/common/core/utils/SystemToolUtils.java
package com.common.core.utils;

import android.net.Uri;
import android.text.TextUtils;

import java.util.regex.Pattern;

/**
 * @author liulun
 * @version V1.0
 * @Description: 系统工具
 * @date 2016/12/2 10:05
 */
public class SystemToolUtils {
    public static boolean isPhoneNumber(String str) {
        if (TextUtils.isEmpty(str)) return false;
        if (str.length() != 11) return false;
        return true;
    }

    public static boolean isChinese(String str) {
        return Pattern.matches("^[\u4E00-\u9FA5]+$", str);
    }

    public static boolean isDouble(String str) {
        try {
            Double.valueOf(str);
            return true;
        } catch (Exception e) {
            return false;
        }
    }

    public static float getFloat(String str) {
        try {
            return Float.valueOf(str);
        } catch (Exception e) {
            return 0;
        }
    }

    public static float getOneFloat(float num) {
        return (float) ((int) (num * 10)) / 10;
    }

    public static <T> T getValue(Class<T> clazz, String value) {
        if (clazz == String.class) {
            return (T) String.valueOf(value);
        } else if (clazz == Byte.class || clazz == byte.class) {
            return (T) Byte.valueOf(value);
        } else if (clazz == Short.class || clazz == short.class) {
            return (T) Short.valueOf(value);
        } else if (clazz == Integer.class || clazz == int.class) {
            return (T) Integer.valueOf(value);
        } else if (clazz == Long.class || clazz == long.class) {
            return (T) Long.valueOf(value);
        } else if (clazz == Float.class || clazz == float.class) {
            return (T) Float.valueOf(value);
        } else if (clazz == Double.class || clazz == double.class) {
            return (T) Double.valueOf(value);
        } else if (clazz == Boolean.class || clazz == boolean.class) {
            return (T) Boolean.valueOf(value);
        } else {
            return (T) value;
        }
    }

    /**
     * 是否金额
     *
     * @return
     */
    public static boolean isDecimalMoney(String str) {
        return Pattern.matches("^\\d+(\\.\\d{1,2})?$", str);
    }

    public static boolean isInteger(String str) {
        return Pattern.matches("^\\d+$", str);
    }

    public static boolean isEqual(String text1, String text2) {
        if (text1 == null) {
            return text2 == null;
        } else {
            return text1.equals(text2);
        }
    }

    public static boolean isHttp(String str) {
        if (str == null) {
            return false;
        } else {
            return str.startsWith("http");
        }
    }

    public static boolean isFileUri(String str) {
        if (str == null) {
            return false;
        } else {
            return str.startsWith("file://");
        }
    }

    public static String getFilePathFromFileUri(String uriStr) {
        if (uriStr == null) {
            return null;
        } else {
            return uriStr.replaceFirst("file://", "");
        }
    }
}

File: example/commons-core/src/main/java/com/common/core/utils/SystemToolUtils.kt
package com.common.core.utils

import android.text.InputFilter
import android.text.Spanned
import android.text.TextUtils
import android.widget.EditText

import java.util.regex.Matcher
import java.util.regex.Pattern

/**
 * @author liulun
 * @version V1.0
 * @Description: 系统工具
 * @date 2016/12/2 10:05
 */
object SystemToolUtils {
    fun isPhoneNumber(str: String): Boolean {
        if (TextUtils.isEmpty(str)) return false
        return if (str.length != 11) false else true
    }

    fun isChinese(str: String): Boolean {
        return Pattern.matches("^[\u4E00-\u9FA5]+$", str)
    }

    fun isDouble(str: String): Boolean {
        try {
            java.lang.Double.valueOf(str)
            return true
        } catch (e: Exception) {
            return false
        }

    }

    fun getFloat(str: String): Float {
        try {
            return java.lang.Float.valueOf(str)!!
        } catch (e: Exception) {
            return 0f
        }

    }

    fun getOneFloat(num: Float): Float {
        return (num * 10).toInt().toFloat() / 10
    }

    fun <T> getValue(clazz: Class<T>, value: String): T {
        return if (clazz == String::class) {
            value.toString() as T
        } else if (clazz == Byte::class|| clazz == Byte::class) {
            java.lang.Byte.valueOf(value) as T
        } else if (clazz == Short::class|| clazz == Short::class) {
            java.lang.Short.valueOf(value) as T
        } else if (clazz == Int::class|| clazz == Int::class) {
            Integer.valueOf(value) as T
        } else if (clazz == Long::class|| clazz == Long::class) {
            java.lang.Long.valueOf(value) as T
        } else if (clazz == Float::class|| clazz == Float::class) {
            java.lang.Float.valueOf(value) as T
        } else if (clazz == Double::class|| clazz == Double::class) {
            java.lang.Double.valueOf(value) as T
        } else if (clazz == Boolean::class|| clazz == Boolean::class) {
            java.lang.Boolean.valueOf(value) as T
        } else {
            value as T
        }
    }

    /**
     * 是否金额
     *
     * @return
     */
    fun isDecimalMoney(str: String): Boolean {
        return Pattern.matches("^\\d+(\\.\\d{1,2})?$", str)
    }

    fun isInteger(str: String): Boolean {
        return Pattern.matches("^\\d+$", str)
    }

    fun isEqual(text1: String?, text2: String?): Boolean {
        return if (text1 == null) {
            text2 == null
        } else {
            text1 == text2
        }
    }
}
